doc/api/crypto.md

```diff
@@ -1365,6 +1365,13 @@ The `key` is the raw key used by the `algorithm` and `iv` is an
 [Buffers][`Buffer`], `TypedArray`, or `DataView`s. If the cipher does not need
 an initialization vector, `iv` may be `null`.
 
+Initialization vectors should be unpredictable; ideally, they will be
+cryptographically random. They do not have to be secret: IVs are typically just
+added to ciphertext messages in plaintext. It may sound contradictory that
+something has to be unpredictable, but doesn’t have to be secret; it’s
+importantto remember that an attacker must not be able to predict ahead of time
```

Comment:
Missing space in `importantto`. Actually, I am not sure what this sentence is supposed to say apart from what the previous said.

Reply:
> It may sound contradictory that something has to be unpredictable, but does not have to be secret; it is important to remember that an attacker must not be able to predict ahead of time
what a given IV will be.

Emphasis on *ahead of time* here. While the IV does not have to be (and mostly isn't) secret, It needs to be unpredictable. It can be as obvious as it could be in hindsight, but it's important that the attack should not be able to predict it *ahead of time*.

I hope that makes my stance clearer? I'm willing to chop this down if you don't find it up to the mark?

Reply:
No strong opinion here, I am okay with or without  the sentence. (Just remember that it is not entirely true for all block cipher modes.)

Reply:
My personal standard here is that in case of ambiguity (here: the exact block cipher being used), we should go with the most secure solution which would work well with *all* of the alternatives, and I think the current definition works for *all* of the ciphers we're supporting.

That said, my knowledge of both the `crypto` subsystem and the Node crypto ecosystem are substantially less than yours, and I'd trust your instincts on this.